### PR TITLE
Feature/stale actions workflow

### PR DESCRIPTION
**Pull request by jaredweinfurtner**
[Remove stale GitHub issues after n days #22](https://github.com/carbynestack/carbynestack/issues/22)

Signed-off-by: Jared Weinfurtner <jared.weinfurtner@de.bosch.com>

### CODE REVIEW

**Comment by jaredweinfurtner** [comment]
adding in auto-labelling functionality...

File: .github/workflows/stale-actions.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+jobs:
+  stale:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/stale@v5
+        with:
+          stale-issue-message: 'This issue or PR has been marked stale because it has been open for 90 days with no activity. This issue or PR will be automatically closed in 5 days if no further activity occurs.'
+          days-before-stale: 90
+          days-before-close: 5
```
Comment (strieflin) [inline .github/workflows/stale-actions.yml]: This should be 30 days.

File: .github/workflows/stale-actions.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+          exempt-issue-labels: 'needs-triage,priority/backlog,priority/critical-urgent,triage/needs-information,priority/awaiting-more-evidence,lifecycle/frozen,lifecycle/stale,lifecycle/rotten,triage/unresolved,triage/duplicate,triage/not-reproducible,triage/accepted,priority/important-longterm,priority/important-soon,kind/documentation,kind/feature,kind/cleanup,kind/bug,kind/support,help-wanted'
```
Comment (strieflin) [inline .github/workflows/stale-actions.yml]: The only exempted label should be `lifecycle/frozen`.

File: .github/workflows/stale-actions.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+on:
+
+  # Allows workflow to be reusable.  @see https://docs.github.com/en/actions/using-workflows/reusing-workflows
+  workflow_call:
+
+jobs:
+  stale:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/stale@v5
```
Comment (strieflin) [inline .github/workflows/stale-actions.yml:17]: This doesn't add the right labels. We should include the following configuration:

1. [stale-issue-label](https://github.com/actions/stale#stale-issue-label) and [stale-pr-label](https://github.com/actions/stale#stale-pr-label) -> `lifecycle/stale`
2. [close-issue-label](https://github.com/actions/stale#close-issue-label) and [close-pr-label](https://github.com/actions/stale#close-pr-label) -> `lifecycle/rotten`
3. [exempt-issue-labels](https://github.com/actions/stale#exempt-issue-labels) and [exempt-pr-labels](https://github.com/actions/stale#exempt-pr-labels) -> `lifecycle/frozen`

Comment (strieflin) [reply .github/workflows/stale-actions.yml:17]: Done

**Review by strieflin — APPROVED** [review]

**Review by sbckr — APPROVED** [review]
LGTM

**Pull request merged**